src/error.rs

```diff
@@ -28,6 +29,12 @@ pub enum Error {
     InvalidMsgId,
     #[fail(display = "Watch folder not found {:?}", _0)]
     WatchFolderNotFound(String),
+    #[fail(display = "Inalid Email: {:?}", _0)]
+    MailParseError(#[cause] mailparse::MailParseError),
+    #[fail(display = "Building invalid Email: {:?}", _0)]
+    LettreError(#[cause] lettre_email::error::Error),
+    #[fail(display = "FromStr error: {:?}", _0)]
+    FromStr(#[cause] mime::FromStrError),
```

Comment:
according to latest discussions and PRs from @link2xt @flub and me we put errors into each module (see sql/smtp and imap.rs).  but that doesn't need to be part of this PR now.